content/blogposts/2022/code-intelligence-platform.md

```diff
@@ -0,0 +1,38 @@
+
+On top of this, devs are more important to their companies than ever. Devs have gone from being viewed as cost centers to being seen as the center of product development, so the issue of complexity is a business problem just as much as a dev problem.
+
+Code search is helpful in tackling complexity, but it's not enough. Code search is for finding things *in* code (matching lines in files). What helps you learn things *about* code? Google went from helping you find things *on* the web (matching web pages), to helping you learn things *from* the web (information and answers), and we're doing the same with code.
+
+This is what we call a code intelligence platform, and compared to code search, it's more helpful, more times per day. **A code intelligence platform helps devs learn and understand information about their code so they can operate more effectively, with a deeper understanding of their codebase and how it works.** 
+
+A code intelligence platform should persist data beyond the code itself and make that data accessible. Here are some examples of what we see the platform doing in the future:
+
+- Trying to figure out the best way to call a function? Code search could show you a semi-exhaustive list of call sites. Code intelligence is also being able to see how the code owners of that function themselves have recently been calling that function.
+- Need to debug an issue in code? Code search could help you navigate within the same project to understand it. Code intelligence is also being able to see what changed and why, and trace calls across projects and dependencies/dependents.
+- Want to fix or improve something across your entire codebase? Code search might help you find where needs fixing. Code intelligence is also being able to push up a quick fix everywhere, track its progress, and monitor for regressions. Sourcegraph does this today.
+- Need to see how your team's API is being used? Code search could show you a list of file locations. Code intelligence is also seeing which other teams are using it and the trend of usage over time.
+
+A code intelligence platform includes code search *and* information about your code from so many other tools, etc., that you rely on: code ownership (and mapping of that to teams, on-call, etc.), code history and reviews, the dependency graph, runtime service/API usage, code and security scanners, deployment status, CI pipeline output, etc. Sourcegraph is committed to being universal so you can get full code intelligence *and* keep using the best tools in each category.
+
+Today, we're announcing Sourcegraph 4.0, the code intelligence platform for devs to unblock themselves and code better. We've spent 9 years building the underlying foundation of code search to be fast, secure, and scalable across ~10^11 lines of code for the world's most demanding software teams, including 4 of 5 FAANG companies and 4 of the top 10 U.S. banks. 
```

Comment:
Demanding? Leading? Strongest? Most advanced?

Reply:
@amieroth How do you feel about how we describe these? Should we add specific customers?

Reply:
A couple things: 
- Let's say: "for the world's most demanding software teams, from 4 of 5 FAANG companies and 4 of the top 10 U.S. banks, to companies launching satellites into space and the leaders in autonomous transportation."
- We wait a long time before getting to the 4.0 part of this, which is the "why now"... can we work in a brief mention of 4.0 in the first paragraph, and then get philosophical, and then get into the stuff here? 
- Also pet peeve, and I know this is a Quinn line, but "code better" doesn't sound like proper English. Can we say "write better code"? Tell me if I'm wrong here if this is standard talk.

Reply:
I don't really like the sound of "code better" either but I think "code better" and "write better code" have different meanings. "Code better" is about being better enabled to write code / be in flow. "Write better code" implies that code quality needs to get better.